### PR TITLE
feat(api): add support for translatable component fallbacks (#863)

### PR DESCRIPTION
**Pull request by NoahvdAa**
Adds support for the new translatable component fallbacks, as requested in #863.

Wasn't sure what to do for MiniMessage, since adding the fallback string in front would break existing configs, and at the end isn't possible because the tag allows for infinite arguments.

adding the fallback to all the static methods was a pain lol

### CODE REVIEW

**Comment by rymiel** [comment]
How does this interact with 2ed1e7c00adda8a0e8bf8fd4e8e86c6c03b8e4fa?

**Comment by NoahvdAa** [comment]
It does not, I didn't see that branch existed :/

**Comment by NoahvdAa** [comment]
Rebased to include kashike's changes.

File: api/src/main/java/net/kyori/adventure/text/TranslatableComponent.java
Comment (qixils) [inline api/src/main/java/net/kyori/adventure/text/TranslatableComponent.java]:
```suggestion
   * @return a translatable component
```

**Comment by NoahvdAa** [comment]
Added MiniMessage support. I went with the separate tag route after some discussion on Discord.

File: api/src/main/java/net/kyori/adventure/text/Component.java
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/text/Component.java]: Any new API should just really do a varargs/iterable of StyleBuilderApplicables - that way it can be done in just one method rather than the bunch that exist here

File: api/src/main/java/net/kyori/adventure/text/TranslatableComponent.java
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/text/TranslatableComponent.java:132]: p much everything relating to this fallback feature needs an `@sinceMinecraft` annotation

File: text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/TranslatableFallbackTag.java
Comment (kezz) [inline text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/TranslatableFallbackTag.java:77]: Does the other tag need to _not_ claim components that do have a fallback set?

File: text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/ComponentSerializerImpl.java
Comment (kezz) [inline text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/ComponentSerializerImpl.java:141]: side note for any mergers: might be worth waiting until the json-serializer merge before merging this)

**Review by kezz — CHANGES_REQUESTED** [review]

File: text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/TranslatableFallbackTag.java
Comment (zml2008) [reply text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/TranslatableFallbackTag.java:77]: I think that's registration order dependent, so for safety that would be a good choice.

File: api/src/main/java/net/kyori/adventure/text/Component.java
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/text/Component.java]: Should this be an iterable of SBAs instead?

File: api/src/main/java/net/kyori/adventure/text/TranslatableComponent.java
Comment (kezz) [inline api/src/main/java/net/kyori/adventure/text/TranslatableComponent.java]: isn't this fallback nullable?

File: text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/StandardTags.java
Comment (kezz) [inline text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/StandardTags.java]:
```suggestion
   * @since 4.13.0
```

**Review by kezz — DISMISSED** [review]

File: text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/ComponentSerializerImpl.java
Comment (zml2008) [reply text-serializer-gson/src/main/java/net/kyori/adventure/text/serializer/gson/ComponentSerializerImpl.java:141]: I think we're going to merge this one first just to avoid rebasing external things unnecessarily

**Review by zml2008 — APPROVED** [review]

**Review by kashike — APPROVED** [review]

**Pull request merged**
